src/System.Text.Primitives/System/Text/Parsing/InvariantUtf16/InvariantUtf16_bool.cs

```diff
@@ -12,31 +12,88 @@ public static partial class InvariantUtf16
         {
             public unsafe static bool TryParseBoolean(char* text, int length, out bool value)
             {
-                int consumed;
-                var span = new ReadOnlySpan<byte>(text, length * 2);
-                return PrimitiveParser.TryParseBoolean(span, out value, out consumed, EncodingData.InvariantUtf16);
+                ReadOnlySpan<char> span = new ReadOnlySpan<char>(text, length);
+                return TryParseBoolean(span, out value);
             }
             public unsafe static bool TryParseBoolean(char* text, int length, out bool value, out int charactersConsumed)
             {
-                var span = new ReadOnlySpan<byte>(text, length * 2);
-                int bytesConsumed;
-                bool result = PrimitiveParser.TryParseBoolean(span, out value, out bytesConsumed, EncodingData.InvariantUtf16);
-                charactersConsumed = bytesConsumed / 2;
-                return result;
+                ReadOnlySpan<char> span = new ReadOnlySpan<char>(text, length);
+                return TryParseBoolean(span, out value, out charactersConsumed);
             }
             public static bool TryParseBoolean(ReadOnlySpan<char> text, out bool value)
             {
-                int consumed;
-                var byteSpan = text.Cast<char, byte>();
-                return PrimitiveParser.TryParseBoolean(byteSpan, out value, out consumed, EncodingData.InvariantUtf16);
+                value = default(bool);
+                if (text.Length < 1)
+                {
+                    return false;
+                }
+
+                char firstCodeUnit = text[0];
+
+                if (firstCodeUnit == '1')
```

Comment:
Why do we have this? Is it because .NET Framework has this behavior? If yes, we should think whether we want to continue with this behavior. It just adds more branches and I am not sure it's useful.